src/dav.js

```diff
@@ -0,0 +1,372 @@
+const { createClient } = require('webdav')
+const parser = require('./xmlParser.js')
+
+export class Dav {
+  constructor (baseUrl, baseUrlv2) {
+    this._XML_CHAR_MAP = {
+      '<': '&lt;',
+      '>': '&gt;',
+      '&': '&amp;',
+      '"': '&quot;',
+      "'": '&apos;'
+    }
+
+    this.baseUrl = baseUrl
+
+    this.userName = null
+
+    this.password = null
+
+    this.xmlNamespaces = {
+      'DAV:': 'd',
+      'http://owncloud.org/ns': 'oc'
+    }
+
+    this.xmlNamespacesComponents = {
+      d: 'DAV:',
+      oc: 'http://owncloud.org/ns'
+    }
+
+    this.client = createClient(baseUrl, {})
+    this.clientv2 = createClient(baseUrlv2, {})
+  }
+
+  _escapeXml (s) {
+    return s.replace(/[<>&"']/g, function (ch) {
+      return this._XML_CHAR_MAP[ch]
+    })
+  };
+
+  /**
+   * Generates a propFind request.
+   *
+   * @param {string} url Url to do the propfind request on
+   * @param {Array} properties List of properties to retrieve.
+   * @param {string} depth "0", "1" or "infinity"
+   * @param {Object} [headers] headers
+   * @return {Promise}
+   */
+  propFind (path, properties, depth = '0', headers = {}, options = {}) {
+    // depth header must be a string, in case a number was passed in
+    depth = '' + depth
+
+    headers = headers || {}
+
+    headers.Depth = depth
+    headers['Content-Type'] = 'application/xml; charset=utf-8'
+
+    var body =
+          '<?xml version="1.0"?>\n' +
+          '<d:propfind '
+    var namespace
+    for (namespace in this.xmlNamespaces) {
+      body += ' xmlns:' + this.xmlNamespaces[namespace] + '="' + namespace + '"'
+    }
+    body += '>\n' +
+          '  <d:prop>\n'
+
+    for (var ii in properties) {
+      // eslint-disable-next-line no-prototype-builtins
+      if (!properties.hasOwnProperty(ii)) {
```

Comment:
```suggestion
      if (!Object.prototype.hasOwnProperty.call(properties, ii)) {
```
if you do it like this you can get rid of the eslint disable command